### PR TITLE
Fix issue #6300: Improve error message for unregistered callbacks.

### PR DESCRIPTION
**Pull request by agiron123**
Fix for issue #6300:
Motivation: When more than one callback is registered to a native module, the error message that a user receives is not indicative of what is really happening.

Test Plan:
MessageQueue. __invokeCallback(cbID, args) will need tests added.

Will need to add appropriate mocks for this._remoteMethodTable to return null and undefined.

Will need to test the cases when debug && this._remoteMethodTable[debug[0]][debug[1]];
returns null, undefined, and a non-null value.

Finally, will need to ensure that invariant(callback, errorMessage) function is called with the appropriate parameters.

As of right now, the code does not have unit tests attached. I can add them later if needed.

Demonstrate the code is solid:
To fix this issue, I first reproduced the conditions of the following issue:
https://github.com/facebook/react-native/issues/6286

To do so, I used the Movies sample project.
After successfully reproducing the error, I then followed the stack trace to come to the appropriate fix.
I can post the code to reproduce the issue if needed. It is essentially the same as that of issue #6286.

Let me know if you have any questions or concerns.

Commit Message:
Fix for issue #6300.
This fix adds a more descriptive error message for the error case when
more than one callback is registered to a native function.

### CODE REVIEW

File: Libraries/Utilities/MessageQueue.js [hunk 1]
```diff
@@ -192,9 +192,14 @@ class MessageQueue {
...
+    let errorMessage = `Callback with id ${cbID}: ${module}.${method}() not found`;
+    if (method !== null || method !== undefined) {
+      errorMessage = `The callback ${method}() exists in module ${module}, `
+      + `but only one callback may be registered to a function in a native module.`;
+    }
     invariant(
       callback,
-      `Callback with id ${cbID}: ${module}.${method}() not found`
+      errorMessage
     );
```
Comment (tadeuzagallo) [inline Libraries/Utilities/MessageQueue.js]: Can you move this whole thing (L192-L203) inside an `if (!callback)`? We can save all the array accesses + string concatenation in the common case.

File: Libraries/Utilities/MessageQueue.js [hunk 2]
```diff
@@ -192,9 +192,14 @@ class MessageQueue {
     let debug = this._debugInfo[cbID >> 1];
     let module = debug && this._remoteModuleTable[debug[0]];
     let method = debug && this._remoteMethodTable[debug[0]][debug[1]];
+    let errorMessage = `Callback with id ${cbID}: ${module}.${method}() not found`;
+    if (method !== null || method !== undefined) {
```
Comment (tadeuzagallo) [inline Libraries/Utilities/MessageQueue.js]: `if (method)` ?

Comment (agiron123) [inline Libraries/Utilities/MessageQueue.js]: Good catch. Fixed in my latest commit.

File: Libraries/Utilities/MessageQueue.js (hunk 1 above)
Comment (agiron123) [inline Libraries/Utilities/MessageQueue.js]: Added to code. I was able to surround everything with the if(!callback) statement except for the module and method variable declaration lines, as they are used both in the if statement and after the if statement. I really like this change. Nice and clean.

**Comment by agiron123** [comment]
@tadeuzagallo Any more feedback?

**Comment by tadeuzagallo** [comment]
@agiron123 No, that looks good, sorry for the delay.

@facebook-github-bot shipit

**Pull request closed without merging**